fixed-hash/Cargo.toml

```diff
@@ -1,28 +1,35 @@
 [package]
 name = "fixed-hash"
-version = "0.2.5"
+version = "0.3.0-beta.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 license = "MIT"
 homepage = "https://github.com/paritytech/parity-common"
 repository = "https://github.com/paritytech/parity-common"
-description = "Fixed-size hashes"
+description = "Macros to define custom fixed-size hash types"
+documentation = "https://docs.rs/fixed-hash/"
+
+[package.metadata.docs.rs]
+features = ["heapsize-support", "quickcheck-support", "api-dummy"]
 
 [dependencies]
 heapsize = { version = "0.4", optional = true }
-rand = { version = "0.4", optional = true }
-rustc-hex = { version = "2.0", optional = true }
-quickcheck = { version = "0.6", optional = true }
-uint = { version = "0.4", path = "../uint", optional = true, default-features = false }
-serde = { version = "1.0", optional = true }
-serde_derive = { version = "1.0", optional = true }
+rand = { version = "0.5", optional = true, default-features = false }
+rustc-hex = { version = "2.0", optional = true, default-features = false }
+quickcheck = { version = "0.7", optional = true }
+byteorder = { version = "1.2.7", optional = true, default-features = false }
+static_assertions = "0.2"
 
 [target.'cfg(not(target_os = "unknown"))'.dependencies]
 libc = { version = "0.2", optional = true, default-features = false }
 
 [features]
-default = ["libc"]
-std = ["rustc-hex", "rand"]
-heapsizeof = ["heapsize"]
-impl_quickcheck_arbitrary = ["quickcheck"]
-serialize = ["serde", "serde_derive"]
-uint_conversions = ["uint"]
+default = ["std", "libc", "rand-support", "rustc-hex-support", "byteorder-support"]
+std = ["rustc-hex/std", "rand/std", "byteorder/std"]
```

Comment:
So you say that all of those features should be disabled by default? I can do that but in my opinion it is always a good thing to have the common uses active by default. This is for example `byteorder-support` and I would even say `rustc-hex-support`.

Reply:
No, that's not what I meant. I think that by having `rand/std` you tell cargo to "use the rand feature and use it with the std feature of rand enabled", i.e. `default = ["std"]` implies that `rustc-hex`, `rand` and `byteorder` will all be enabled (and will all be configured to use their respective `std` feature). Do double-check me on that though, I'm not 100% sure!

Reply:
The problem is that this requires not having `rand`, `byteorder` and `rustc-hex` being activated but their features `rand-support`, `byteorder-support` and `rustc-hex-support` should be enabled since the entire code is guarded by them and not by their respective features which is more accurate I think. So this would cause problems I think.

Reply:
Btw.: I think the normal use case is to either use `std` feature or `libc` but not both at the same time. So implying the one by the other seems to be wrong to me. But I might be confused about that!